A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -1924,13 +1924,18 @@ class FMCMainDisplay extends BaseAirliners {
         return true;
     }
 
-    getTransitionAltitude() {
-        if (isFinite(this.transitionAltitude)) {
-            return this.transitionAltitude;
-        }
-
-        // TODO fetch this from the nav database once we have it in future
-        return 10000;
+    async getTransitionAltitude(departICAO) {
+        await NXApi.getAirport(departICAO)
+            .then((data) => {
+                this.transitionAltitude = data.transAlt;
+                if (this.transitionAltitude === -1) {
+                    console.log("NO DEPARTURE TA");
+                    return 10000;
+                } else {
+                    SimVar.SetSimVarValue("L:AIRLINER_TRANS_ALT", "Number", this.transitionAltitude);
+                    return this.transitionAltitude;
+                }
+            })
```

Comment:
No reason for `.then` if you're async anyway.

```typescript
        const data = await NXApi.getAirport(departICAO);
        this.transitionAltitude = data.transAlt;
        if (this.transitionAltitude === -1) {
            console.log("NO DEPARTURE TA");
            return 10000;
        } else {
            SimVar.SetSimVarValue("L:AIRLINER_TRANS_ALT", "Number", this.transitionAltitude);
            return this.transitionAltitude;
        }
```

However, don't just copy this, there's more issues:
- We probably always want to set L:AIRLINER_TRANS_ALT, even if there is no data, to make sure the sim is up-to-date.
- What happens when `NXApi.getAirport` rejects, maybe because our server is down or because the user has no internet or is behind a firewall? We should add try/catch around it and ensure we never crash but always end up with a valid value.
- You're converting this function to be asynchronous. Where is it used? Do you need to add a callback somewhere else?

Reply:
1. current working on it. Benjozork said, using popular airport data, only insert that on sim and others coming from api.
2. yes if rejected like 1. currently just do return 10000 which is default value on sim
3. i thought await need async so i did that.

Reply:
> i thought await need async so i did that.

That is correct. However, `getTransitionAltitude` is currently probably called from another place, that is expecting the return value of `getTransitionAltitude` to be a number. But by making it an `async` function it's actually returning a Promise that returns a number. This means, any other function that calls this will need to be converted to `async` + `await getTransitionAltitude` itself, or you need to rewrite them to use `getTransitionAltitude().then(altitude => { /* ... */ })`.

Reply:
i changed that call only one time when init & changing airport(same as add airport) so i think it is okay :)